main.tf

```diff
@@ -16,11 +16,15 @@ data "aws_caller_identity" "current" {}
 
 resource "aws_kms_key" "datadog" {
   description         = "KMS key for datadog lambda"
-  policy              = data.aws_iam_policy_document.kms_key_policy.json
   enable_key_rotation = true
   tags                = local.tags
 }
 
+resource "aws_kms_key_policy" "datadog" {
```

Comment:
If we try to adjust an existing policy, we get cycle errors unless we use this separate resource.